sale_order_price_recalculation/tests/test_sale_order_price_recalculation.py

```diff
@@ -0,0 +1,103 @@
+                            "applied_on": "3_global",
+                            "compute_price": "formula",
+                            "base": "list_price",
+                        },
+                    ),
+                ],
+            }
+        )
+        self.pricelist_dto = self.env["product.pricelist"].create(
+            {
+                "name": "Test pricelist discount policy",
+                "discount_policy": "without_discount",
+                "item_ids": [
+                    (
+                        0,
+                        0,
+                        {
+                            "applied_on": "1_product",
+                            "compute_price": "percentage",
+                            "percent_price": 50.0,
+                            "product_tmpl_id": self.product.product_tmpl_id.id,
+                        },
+                    ),
+                ],
+            }
+        )
+        self.sale_order = self.env["sale.order"].create(
+            {
+                "partner_id": self.partner.id,
+                "partner_invoice_id": self.partner.id,
+                "partner_shipping_id": self.partner.id,
+                "pricelist_id": self.pricelist.id,
+            }
+        )
+        line_vals = {
+            "product_id": self.product.id,
+            "name": self.product.name,
+            "product_uom_qty": 1.0,
+            "product_uom": self.product.uom_id.id,
+            "price_unit": self.product.lst_price,
+            "order_id": self.sale_order.id,
+        }
+        self.sale_order_line = self.env["sale.order.line"].create(line_vals)
+
+    def test_price_recalculation(self):
+        # Check current price
+        self.sale_order_line.name = "My product description"
+        self.assertEqual(self.sale_order_line.price_unit, self.product.lst_price)
+        # Change price
+        self.product.lst_price = 500
+        # Launch recalculation
+        self.sale_order.recalculate_prices()
+        # Check if the price has been updated
+        self.assertEqual(self.sale_order_line.price_unit, self.product.lst_price)
+        # Check if quantities have changed
+        self.assertEqual(self.sale_order_line.product_uom_qty, 1.0)
+        # Check the description still unchanged
+        self.assertEqual(self.sale_order_line.name, "My product description")
+        # Apply listprice with a discount
+        self.sale_order.pricelist_id = self.pricelist_dto
+        self.sale_order.recalculate_prices()
+        self.assertEqual(self.sale_order_line.discount, 50.0)
+
+    def test_name_recalculation(self):
+        self.sale_order_line.price_unit = 150.0
+        initial_price = self.sale_order_line.price_unit
+        self.assertEqual(self.sale_order_line.name, self.product.name)
+        self.sale_order_line.name = "Custom Jacket"
+        self.sale_order.recalculate_names()
+        self.assertEqual("Jacket - Color: Black - Size: XL", self.sale_order_line.name)
```

Comment:
@rousseldenis Take into account that if the product has a default_code the name would be "[00001]......" so the test fails...
If product_sequence is installed that always creates a product ref this test fails
Maybe the best option is do the test with a product with default_code

Reply:
@sergio-teruel I didn't even touched to tests because for a migration, tests should pass as is.

Reply:
@rousseldenis what my colleague is talking about is the resilience of the tests when combined with other OCA modules, as our CI reveals. In this case, combining this module with `product_sequence`. Indeed, the tests weren't resilient before and it's not your fault. But the proposal is to change it for that. I propose an even more resilient option, as if you have the module `sale_line_description`, Sergio's proposal will fail as well:

```suggestion
        self.assertIn("Jacket - Color: Black - Size: XL", self.sale_order_line.name)
```

Reply:
Or even a better one:
```suggestion
        self.assertNotEqual("Custom Jacket", self.sale_order_line.name)
```
the test is satisfied as it checks that the description has been recomputed, no matter to what value.

Reply:
Ok agree with proposal, makes sense. I need time to get back on this, but if you want to make a PR on my branch, don't hesitate.

Reply:
I can supersede the PR if you want, but would you accept a PR removing the dependency on `onchange_helper` and switching to `Form`?

Reply:
Yes, do it.